jvm/hermes/src/main/jni/JJSIValue.h

```diff
@@ -129,7 +126,14 @@ class JJSIValue : public JJSIValueHybridClass {
     static local_ref<jhybridobject> createFromJsonUtf8(alias_ref<jclass>, alias_ref<JRuntimeThreadContext>, alias_ref<JJSIRuntime::jhybridobject> jRuntime, alias_ref<JByteBuffer> json);
     static bool strictEquals(alias_ref<jclass>, alias_ref<JRuntimeThreadContext>, alias_ref<JJSIRuntime::jhybridobject> jRuntime, alias_ref<jhybridobject> a, alias_ref<jhybridobject> b);
 
-    explicit JJSIValue(Value&& value) : HybridClass(), value_(std::make_unique<Value>(std::move(value))) {}
+    explicit JJSIValue(std::shared_ptr<RuntimeScope> scope, Value&& value) : HybridClass(), scope_(scope) {
+        // internally creates unique ptr
+        scope->trackValue(this, std::move(value));
+    }
```

Comment:
If/when there is a singular API to track arbitrary runtime types, we should try to capture this as a common _super_ constructor for `JJSIValueHybridClass` to enforce for any subclass (there are some new ones that may exist in the future).